vortex-array/src/array/primitive/accessor.rs

```diff
@@ -1,25 +1,30 @@
+use std::iter;
+
 use vortex_dtype::NativePType;
 use vortex_error::VortexResult;
 
 use crate::accessor::ArrayAccessor;
 use crate::array::primitive::PrimitiveArray;
-use crate::validity::ArrayValidity;
+use crate::validity::Validity;
+use crate::{ArrayLen, IntoArrayVariant};
 
 impl<T: NativePType> ArrayAccessor<T> for PrimitiveArray {
     fn with_iterator<F, R>(&self, f: F) -> VortexResult<R>
     where
         F: for<'a> FnOnce(&mut (dyn Iterator<Item = Option<&'a T>>)) -> R,
     {
-        match self.logical_validity().to_null_buffer()? {
-            None => {
+        match self.validity() {
```

Comment:
this is a slight drive by to avoid allocating all null buffers though I doubt it happens often in practice